Sources/Vehicle/Extensions/HAEntityExtension.swift

```diff
@@ -0,0 +1,215 @@
+import Foundation
+import HAKit
+import PromiseKit
+import Shared
+import SwiftUI
+
+extension HAEntity {
+    public static func getIconForDomain(domain: String, size: CGSize) -> UIImage? {
+        do {
+            let tmpEntity = try HAEntity(entityId: "\(domain).ha_ios_placeholder",
```

Comment:
this is a bit hacky; you can create a domain enum to handle this part for the defaults rather than a constructed entity in the middle.

Reply:
This was done as it was done on Android. As the domain icon is based on the way the entity icon is obtained, this entity is created to obtain the icon that will be used as the domain icon.

[https://github.com/home-assistant/android/blob/1162a82c4896f74d7d9816082866f2343e4e31c5/app/src/main/java/io/homeassistant/companion/android/util/vehicle/GridItems.kt#L132]

Reply:
I moved the logic to return the domain icon to the Domain enum